python/tvm/micro/testing/pytest_plugin.py

```diff
@@ -0,0 +1,126 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: disable=invalid-name,redefined-outer-name
+""" microTVM testing fixtures used to deduce testing argument
+    values from testing parameters """
+
+import pathlib
+import json
+import os
+import datetime
+import pytest
+
+import tvm
+from tvm.contrib.utils import tempdir
+
+
+def zephyr_boards() -> dict:
+    """Returns a dict mapping board to target model"""
+    with open(
+        pathlib.Path(tvm.micro.get_microtvm_template_projects("zephyr")) / "boards.json"
+    ) as f:
+        board_properties = json.load(f)
+
+    boards_model = {board: info["model"] for board, info in board_properties.items()}
+    return boards_model
+
+
+def get_boards(platform: str) -> dict:
+    """Returns a dict mapping board to target model"""
+    with open(
+        pathlib.Path(tvm.micro.get_microtvm_template_projects(platform)) / "boards.json"
+    ) as f:
+        board_properties = json.load(f)
+
+    boards_model = {board: info["model"] for board, info in board_properties.items()}
+    return boards_model
+
+
+def pytest_addoption(parser):
+    """Adds more pytest arguments"""
+    parser.addoption(
+        "--board",
```

Comment:
Love this change! It has been a long time in the workds.

Reply:
We should also document explicitly the difference between `platform`s and `board`s in our usage - it might not be obvious to others.

Reply:
added more details